src/zenml/deployers/base_deployer.py

```diff
@@ -0,0 +1,996 @@
+
+            # Subtract the time spent deploying the deployment from the timeout
+            timeout = timeout - int(time.time() - start_time)
+            deployment, _ = self._poll_deployment(
+                deployment, DeploymentStatus.RUNNING, timeout
+            )
+
+            if deployment.status != DeploymentStatus.RUNNING:
+                raise DeploymentProvisionError(
+                    f"Failed to deploy deployment {deployment.name}: "
+                    f"The deployment's operational state is {deployment.status}. "
+                    "Please check the status or logs of the deployment for more "
+                    "information."
+                )
+
+            analytics_handler.metadata = (
+                self._get_deployment_analytics_metadata(
+                    deployment=deployment,
+                    stack=stack,
+                )
+            )
+
+            return deployment
+
+    def refresh_deployment(
+        self,
+        deployment_name_or_id: Union[str, UUID],
+        project: Optional[UUID] = None,
+    ) -> DeploymentResponse:
+        """Refresh the status of a deployment by name or ID.
+
+        Call this to refresh the operational state of a deployment.
+
+        Args:
+            deployment_name_or_id: The name or ID of the deployment to get.
+            project: The project ID of the deployment to get. Required
+                if a name is provided.
+
+        Returns:
+            The deployment.
+
+        Raises:
+            DeploymentNotFoundError: if the deployment is not found.
+            DeployerError: if an unexpected error occurs.
+        """
+        client = Client()
+        try:
+            deployment = client.get_deployment(
+                deployment_name_or_id, project=project
+            )
+        except KeyError:
+            raise DeploymentNotFoundError(
+                f"Deployment with name or ID '{deployment_name_or_id}' "
+                f"not found"
+            )
+
+        self._check_deployment_deployer(deployment)
+
+        deployment_state = DeploymentOperationalState(
+            status=DeploymentStatus.ERROR,
+        )
+        try:
+            deployment_state = self.do_get_deployment(deployment)
+        except DeploymentNotFoundError:
+            deployment_state.status = DeploymentStatus.ABSENT
+            deployment = self._update_deployment(deployment, deployment_state)
+        except DeployerError as e:
+            self._update_deployment(deployment, deployment_state)
+            raise DeployerError(
+                f"Failed to refresh deployment {deployment_name_or_id}: {e}"
+            ) from e
+        except Exception as e:
+            self._update_deployment(deployment, deployment_state)
+            raise DeployerError(
+                f"Unexpected error while refreshing deployment for "
+                f"{deployment_name_or_id}: {e}"
+            ) from e
+
+        return deployment
```

Comment:
In the success case, this returns the deployment without actually updating its operational state as far as I can see? Not sure if this is intended or not, but seems weird?